packages/esbuild-plugin/src/index.js

```diff
@@ -0,0 +1,176 @@
+      const stylexRules: { [string]: $ReadOnlyArray<Rule> } = {};
+
+      build.onEnd(async ({ outputFiles }: BuildResult<>) => {
+        const rules: Array<Rule> = Object.values(stylexRules).flat();
+
+        if (rules.length === 0) {
+          return;
+        }
+
+        const collectedCSS = stylexBabelPlugin.processStylexRules(
+          rules,
+          useCSSLayers,
+        );
+        const shouldWriteToDisk =
+          build.initialOptions.write === undefined ||
+          build.initialOptions.write;
+
+        if (shouldWriteToDisk) {
+          await mkdir(path.dirname(generatedCSSFileName), {
+            recursive: true,
+          });
+          await writeFile(generatedCSSFileName, collectedCSS, 'utf8');
+
+          return;
+        }
+
+        if (outputFiles !== undefined) {
+          outputFiles.push({
+            path: '<stdout>',
+            contents: new TextEncoder().encode(collectedCSS),
+            hash: utils.hash(new Date().getTime().toString()),
+            get text() {
+              return collectedCSS;
+            },
+          });
+        }
+      });
+
+      build.onLoad(
+        { filter: STYLEX_PLUGIN_ONLOAD_FILTER },
+        async (args: OnLoadArgs) => {
+          const currFilePath = args.path;
+          const inputCode = await readFile(currFilePath, 'utf8');
+
+          if (
+            !stylexImports.some((importName) => inputCode.includes(importName))
+          ) {
+            // avoid transform if file doesn't have stylex imports
+            // esbuild proceeds to the next callback
+            return;
+          }
+
+          const transformResult = await transformAsync(inputCode, {
+            babelrc: false,
+            filename: currFilePath,
+            presets,
+            plugins: [
+              ...plugins,
+              ...getFlowOrTypeScriptBabelSyntaxPlugins(currFilePath),
+              jsxSyntaxPlugin,
+              stylexBabelPlugin.withOptions({
+                ...options,
+                dev,
+                unstable_moduleResolution,
+              }),
+            ],
+          });
+
+          const loader = getEsbuildLoader(currFilePath);
+
+          if (transformResult === null) {
+            console.warn('StyleX: transformAsync returned null');
+            return { code: inputCode, loader };
+          }
+
+          const { code, metadata } = transformResult;
+
+          if (code === null) {
+            console.warn('StyleX: transformAsync returned null code');
+            return { code: inputCode, loader };
+          }
+
+          if (
+            !dev &&
+            (metadata: $FlowFixMe).stylex !== null &&
+            (metadata: $FlowFixMe).stylex.length > 0
+          ) {
+            stylexRules[args.path] = (metadata: $FlowFixMe).stylex;
+          }
+
+          return {
+            contents: code,
+            loader,
+          };
+        },
+      );
+    },
+  };
+}
+
+function getEsbuildLoader(fileName: string) {
```

Comment:
Also, `.mjs` and `.cjs` and `.mts` and `.cts`

Reply:
`ts` loader handles `.mts` and `.cts` files, while `js` loader handles `.cjs` and `.mjs` -- [see here](https://esbuild.github.io/content-types/) for the reference